osquery/events/linux/syslog.h

```diff
@@ -0,0 +1,172 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <fstream>
+#include <stdio.h>
+#include <map>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <gtest/gtest.h>
+
+#include <osquery/events.h>
+
+namespace osquery {
+
+/**
+ * @brief Subscription context for syslog events
+ *
+ * Currently there is no use for the subscription context, so this class
+ * remains empty.
+ */
+
+struct SyslogSubscriptionContext : public SubscriptionContext {
+ private:
+  friend class SyslogEventPublisher;
+};
+
+/**
+ * @brief Event details for SyslogEventPublisher events
+ */
+struct SyslogEventContext : public EventContext {
+  /**
+   * @brief The syslog message tokenized into fields.
+   *
+   * Fields will be stripped of extra space
+   */
+  std::map<std::string, std::string> fields;
+};
+
+using SyslogEventContextRef = std::shared_ptr<SyslogEventContext>;
+using SyslogSubscriptionContextRef = std::shared_ptr<SyslogSubscriptionContext>;
+
+/**
+ * @brief Event publisher for syslog lines forwarded through rsyslog
+ *
+ * This event publisher ingests JSON representations of syslog entries, and
+ * publishes them to it's subscribers. In order for it to function properly,
+ * rsyslog must be configured to forward JSON to a named pipe that this
+ * publisher will read from.
+ */
+class SyslogEventPublisher
+    : public EventPublisher<SyslogSubscriptionContext, SyslogEventContext> {
+  DECLARE_PUBLISHER("syslog");
+
+ public:
+  Status setUp() override;
+
+  void configure() override;
+
+  void tearDown() override;
+
+  Status run() override;
+
+ public:
+  SyslogEventPublisher() : EventPublisher(){};
```

Comment:
This is weird, for some reason clang-format wanted `;`, but it really shouldn't be there and newer versions of GCC/clang will lint warning this.